test/plugin/scenarios/postgresql-scenario/src/main/java/org/apache/skywalking/apm/testcase/postgresql/controller/CaseController.java

```diff
@@ -0,0 +1,88 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.testcase.postgresql.controller;
+
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/postgresql-scenario/case")
+public class CaseController {
+    private Logger logger = LogManager.getLogger(CaseController.class);
+
+    @Autowired
+    PostgresqlConfig postgresqlConfig;
+
+    @GetMapping("/healthcheck")
+    public String healthcheck() throws Exception {
+        SQLExecutor sqlExecute = null;
+        try {
+            sqlExecute = new SQLExecutor(postgresqlConfig);
+            sqlExecute.checkPG(ConstSql.TEST_SQL);
+        } catch (SQLException e) {
+            logger.error("Failed to execute sql.", e);
+            throw e;
+        } finally {
+            if (sqlExecute != null) {
+                try {
+                    sqlExecute.closeConnection();
+                } catch (SQLException e) {
+                    logger.error("Failed to close connection.", e);
+                }
+            }
+        }
+        return "Success";
+    }
+
+    @GetMapping("/postgres")
+    public String postgres() throws SQLException {
+        logger.info("Begin to start execute sql");
+        SQLExecutor sqlExecute = null;
+        try {
+            sqlExecute = new SQLExecutor(postgresqlConfig);
+            sqlExecute.createTable(ConstSql.CREATE_TABLE_SQL);
+            sqlExecute.insertData(ConstSql.INSERT_DATA_SQL, "1", "1");
+            sqlExecute.dropTable(ConstSql.DROP_TABLE_SQL);
+        } catch (SQLException e) {
+            logger.error("Failed to execute sql.", e);
+            throw e;
+        } finally {
+            if (sqlExecute != null) {
+                try {
+                    sqlExecute.closeConnection();
```

Comment:
Is this why two `close` spans are generated? Could you recheck it?

Reply:
refer https://github.com/apache/skywalking/pull/3644#issuecomment-544154295

Reply:
@dmsolr I have debug the code, https://github.com/apache/skywalking/pull/3644#issuecomment-544152900 is right, ConnectionServiceMethodInterceptor in jdbc-commons do the Interceptor, execute twice.

Reply:
you need to remove one, right?